woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -1745,6 +1746,185 @@ def test_sets_koalas_option_on_init(sample_df_koalas):
         assert ks.get_option('compute.ops_on_diff_frames') is True
 
 
+def test_setitem_invalid_input(sample_df):
+    df = sample_df.copy()
+    df.ww.init(index='id', time_index='signup_date')
+
+    error_msg = 'New column must be of Series type'
+    with pytest.raises(ValueError, match=error_msg):
+        df.ww['test'] = [1, 2, 3]
+
+    error_msg = 'Cannot reassign index. Change column name and then use df.ww.set_index to reassign index.'
+    with pytest.raises(KeyError, match=error_msg):
+        df.ww['id'] = df.id
+
+    error_msg = 'Cannot reassign time index. Change column name and then use df.ww.set_time_index to reassign time index.'
+    with pytest.raises(KeyError, match=error_msg):
+        df.ww['signup_date'] = df.signup_date
+
+
+def test_setitem_different_name(sample_df):
+    df = sample_df.copy()
+    df.ww.init()
+
+    new_series = pd.Series([1, 2, 3, 4], name='wrong', dtype='float')
+    if ks and isinstance(sample_df, ks.DataFrame):
+        new_series = ks.Series(new_series)
+
+    warning = 'Name mismatch between wrong and id. Series name is now id'
+    with pytest.warns(ColumnNameMismatchWarning, match=warning):
+        df.ww['id'] = new_series
+
+    assert df.ww['id'].name == 'id'
+    assert 'id' in df.columns
+    assert 'id' in df.ww._schema.columns.keys()
+    assert 'wrong' not in df.ww.columns
```

Comment:
I think I might have been a little confused in my prior comment here.  In any case, let's update these assertions:
`assert 'id' in df.ww.columns` which will check that 'id' is present in the schema dictionary
`asssert 'wrong' not in df.ww.columns` to check that 'wrong' is not present
add `assert 'wrong not in df.columns` just to make sure a column named 'wrong didn't end up in the dataframe somehow.

Reply:
✅ 60e44f8